packages/x-components/tests/unit/main-scroll.spec.ts

```diff
@@ -0,0 +1,264 @@
+  itemHeight = '50px',
+  threshold,
+  margin,
+  minScrollTrigger,
+  useWindow,
+  windowScrollingElement
+}: RenderMainScrollOptions = {}): RenderMainScrollAPI {
+  const userScrolledToElementSpy = cy.spy();
+  XPlugin.bus.on('UserScrolledToElement').subscribe(userScrolledToElementSpy);
+
+  document.body.dataset.test = document.documentElement.dataset.test = '';
+  if (windowScrollingElement === 'body') {
+    document.body.dataset.test = 'scroll';
+  } else if (windowScrollingElement === 'html') {
+    document.documentElement.dataset.test = 'scroll';
+  }
+
+  cy.viewport(1920, 200);
+  mount(
+    {
+      components: {
+        MainScroll,
+        MainScrollItem
+      },
+      template: `
+        <MainScroll v-bind="{ threshold, margin, useWindow, minScrollTrigger }">
+        <div class="container" ${!useWindow ? `data-test="scroll"` : ''}>
+          <MainScrollItem
+            v-for="item in items"
+            class="item"
+            tag="article"
+            :item="item"
+            :data-test="item.id">
+            {{ item.id }}
+          </MainScrollItem>
+        </div>
+        </MainScroll>
+      `,
+      data() {
+        return {
+          items: Array.from({ length: itemsCount }, (_, index) => ({ id: `item-${index}` })),
+          threshold,
+          margin,
+          useWindow,
+          minScrollTrigger
+        };
+      }
+    },
+    {
+      style: `
+        ${windowScrollingElement === 'body' ? 'html { overflow: hidden; }' : ''}
+
+        html,
+        body,
+        #__cy_root {
+          margin: 0;
+          height: 100%;
+          max-height: 100%;
+        }
+
+
+        [data-test='scroll'] {
+          overflow: auto;
+          height: 100%;
+        }
+
+        .item {
+          height: ${itemHeight};
+        }`
+    }
+  );
+  return {
+    scrollToItem(index) {
+      cy.getByDataTest(`item-${index}`).then($0 => $0.get(0).scrollIntoView());
+    },
+    scrollBy(y) {
+      return cy.getByDataTest('scroll').then($0 => {
+        $0.get(0).scrollBy(0, y);
+      });
+    },
+    restoreScroll(index) {
+      XPlugin.bus.emit('ParamsLoadedFromUrl', <UrlParams>{ scroll: `item-${index}` });
+    },
+    getItem(index) {
+      return cy.getByDataTest(`item-${index}`);
+    },
+    getScrollingElement() {
+      return cy.getByDataTest('scroll');
+    },
+    userScrolledToElementSpy: () => cy.wrap(userScrolledToElementSpy)
+  };
+}
+
+before(() => {
+  Vue.use(xPlugin, { adapter: mockedAdapter });
+});
+
+describe('testing MainScroll component', () => {
+  const cases: Array<Partial<RenderMainScrollOptions> & { description: string }> = [
+    { description: 'when using the html element', useWindow: true, windowScrollingElement: 'html' },
+    { description: 'when using the body element', useWindow: true, windowScrollingElement: 'body' },
+    { description: 'when using a custom scrolling element' }
+  ];
+
+  cases.forEach(({ description, ...defaultParameters }) => {
```

Comment:
Before anyone says something, I hate the debugging experience of using this approach, you have to comment first the case, and then flag the test you want to debug with `.only`, but I hate more having to triplicate the tests.

Reply:
maybe this article could help: https://dev.to/tinesoft/be-smart-write-parameterized-tests-with-jest-3eab

Reply:
I didn't know that Jest supported this by default, which is awesome if we need to do this (And I'm planing to do so in the scroll component). However I still have the same complains: I can't trigger a single test scenario without having to comment the rest.

Reply:
https://github.com/facebook/jest/tree/main/packages/jest-each#tagged-template-literal-of-rows